resolver/resolver.go

```diff
@@ -147,6 +147,12 @@ func (a Address) Equal(o Address) bool {
 		a.Type == o.Type && a.Metadata == o.Metadata
 }
 
+// equalWithAttributes is a limited comparison operation where select fields of
+// Address are used to decide if a and o are identical.
```

Comment:
This sounds really arbitrary.  The explanation should say something about how the addresses are equivalent w.r.t. how a subchannel would be created by it.  If any of these fields change, the subchannels created by the addresses would not be equivalent.

Reply:
Removed this function after making the key to be of type `resolver.Address` with only the fields that we care about being set.